packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2890,8 +2984,18 @@ export class ContainerRuntime
 				type === ContainerMessageType.Attach &&
 				this.disableAttachReorder !== true
 			) {
+				// Generate an Id allocation op if compressed Ids have been generated
+				if (idAllocationBatchMessage !== undefined) {
```

Comment:
You can move this to before the if-else block since the same thing is happening in both. You should do this together with the code above that creates the id allocation message. And then move the block of code into its own function to keep the submit function relatively clean.

Reply:
I've created a `generateAndSubmitIdAllocationOp` function